crates/turborepo-lib/src/shim.rs

```diff
@@ -46,6 +46,24 @@ fn turbo_version_has_shim(version: &str) -> bool {
     version.major > 1
 }
 
+fn turbo_version_in_range(version: &str, root_path: &Path) -> bool {
+    let version = Version::parse(version).unwrap();
+
+    let version_req_string = match read_turbo_json(root_path) {
+        Ok(turbo_json) => turbo_json.turboVersion,
+
+        // Either not present or it failed to read `turbo.json`.
+        // We'll assume this is *.
+        Err(_) => "*".to_string(),
+    };
```

Comment:
This is extreme nit and primarily for educational purposes of how we can avoid an allocation if we fail at reading `turbo.json` (the inner `as_str` might be unnecessary, but off the top of my head I can't remember)
```suggestion
    let turbo_json = read_turbo_json(root_path);
    // Either not present or it failed to read `turbo.json`.
    // We'll assume this is *.
    let version_req_string = turbo_json.as_ref().map_or("*", |turbo_json| turbo_json.turboVersion.as_str()).unwrap_or("*");
```

Reply:
By making the whole thing a `Result` this ended up cleaning up automatically.